alacritty/src/window_context.rs

```diff
@@ -81,15 +83,24 @@ impl WindowContext {
         // This object contains all of the state about what's being displayed. It's
         // wrapped in a clonable mutex since both the I/O loop and display need to
         // access it.
-        let terminal = Term::new(config, display.size_info, event_proxy.clone());
+        let terminal = Term::new(&config.terminal_config, display.size_info, event_proxy.clone());
         let terminal = Arc::new(FairMutex::new(terminal));
 
         // Create the PTY.
         //
         // The PTY forks a process to run the shell on the slave side of the
         // pseudoterminal. A file descriptor for the master side is retained for
         // reading/writing to the shell.
-        let pty = tty::new(config, &display.size_info, display.window.x11_window_id());
+        let new_pty_config;
+        let pty_config = match options {
+            Some(options) => {
+                new_pty_config = options.into();
+                &new_pty_config
+            },
+            None => &config.terminal_config.pty_config,
+        };
+
+        let pty = tty::new(pty_config, &display.size_info, display.window.x11_window_id())?;
```

Comment:
After the completions PR, this should be pretty much good to go. Only thing left would be my suggestion to maybe try this code with map/unwrap_or?

Seems like you've got the whole `new_pty_config` temporary variable to work around borrowing issues. Sounds a lot like a `Cow` usecase to me?

Reply:
It's not about borrowing, it's more about picking right reference to variable, not sure how `Cow` (copy on write) relates to that? I mean, I could in theory store Cow then clone it and then `if Some(..) = options` update its variable, but I don't like that approach.

Reply:
It seems like you're declaring the `new_pty_config` variable to extend the lifetime of the `options.into()` since you expect to store a reference because you can't move out of `config.terminal_config`. If you instead return a `Cow` from the match which stores `Owned` in the `Some` case and `Borrowed` in the `None` case you'll avoid having to declare the `new_pty_config` environment variable for creating a temporary borrowed value.